File: 87.py
def minPlatforms(arr, dep):
    arr.sort()
    dep.sort()
    
    i = j = 0
    current = result = 0
    n = len(arr)
    
    while i < n and j < n:
        if arr[i] <= dep[j]:
            current += 1
            result = max(result, current)
            i += 1
        else:
            current -= 1
            j += 1
    return result

# Driver
arr = [900,940,950,1100,1500,1800]
dep = [910,1200,1120,1130,1900,2000]
print("Minimum Platforms:", minPlatforms(arr, dep))

File: 87.js
function minPlatforms(arr, dep) {
    arr.sort((a,b)=>a-b);
    dep.sort((a,b)=>a-b);

    let i = 0, j = 0;
    let current = 0, result = 0;

    while (i < arr.length && j < dep.length) {
        if (arr[i] <= dep[j]) {
            current++;
            result = Math.max(result, current);
            i++;
        } else {
            current--;
            j++;
        }
    }
    return result;
}

// Driver
console.log("Minimum Platforms:",
    minPlatforms([900,940,950,1100,1500,1800],
                 [910,1200,1120,1130,1900,2000])
);
